3-typescript/1-ecommerce/2-products.ts

```diff
@@ -36,10 +85,30 @@ async function analyzeProductPrices(products: any[]): Promise<any> {}
  */
 
 async function buildProductCatalog(
-  products: unknown[],
-  brands: unknown[],
-): Promise<unknown[]> {
-  return [];
+  products: Product[],
+  brands: Brand[],
+): Promise<EnrichedProduct[]> {
+
+  const EnrichedProducts: EnrichedProduct[] = new Array();
+
+  const activeBrands = brands.filter((brand) => brand.isActive);
+
+  const activeProducts = products.filter((product) => product.isActive);
+
+  activeProducts.forEach(product => {
+    const brand = activeBrands.find(brand => brand.id === product.brandId)
+    if (brand) {
+      const { id, isActive, ...brandInfo } = brand;
+      const enrichedProduct: EnrichedProduct = {
+        ...product,
+        brandInfo
```

Comment:
In this way, we're not keeping unused variables. The `BrandInfo` will only show what properties are available, so it will omit the id and isActive properties.
```suggestion
      const enrichedProduct: EnrichedProduct = {
        ...product,
        brandInfo: brand
```